tests/ut/jit/test_jit_compile_extraction.py

```
@@ -374,6 +410,85 @@ def scalar_sig_kernel(
+        assert "pl.RUNTIME" not in prog
+        assert "pl.tile.adds(tile, n)" in prog
+
+    def test_runtime_marker_rejected_on_the_dispatch_path(self):
+        """``pl.RUNTIME`` is a compile-time marker. Binding real arguments — a
+        dispatch call, or a sample-argument compile — must reject it by name
+        instead of letting it reach the runtime as a bogus scalar. The signature
+        default makes this reachable from a plain ``kernel(a, c)`` call."""
+        torch = pytest.importorskip("torch")
+
+        t = torch.zeros(256, 128, dtype=torch.float32)
+        with pytest.raises(TypeError, match="'n' received pl.RUNTIME"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)